CMakeLists.txt

```diff
@@ -0,0 +1,256 @@
+    DESCRIPTION "Mapnik is an open source toolkit for developing mapping applications" 
+    LANGUAGES CXX 
+)
+
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
+
+option(BUILD_SHARED_LIBS "build mapnik dynamic" ON)
+option(BUILD_TEST "builds the tests" ON)
+option(USE_EXTERNAL_MAPBOX_GEOMETRY "Use a external mapnik/geometry.hpp. If off, use the submodule" OFF)
+option(USE_EXTERNAL_MAPBOX_POLYLABEL "Use a external mapnik/polylabel. If off, use the submodule" OFF)
+option(USE_EXTERNAL_MAPBOX_PROTOZERO "Use a external mapnik/protozero. If off, use the submodule" OFF)
+option(USE_EXTERNAL_MAPBOX_VARIANT "Use a external mapnik/variant. If off, use the submodule" OFF)
+option(USE_JPEG "adds jpeg support" ON)
+option(USE_PNG "adds png support" ON)
+option(USE_TIFF "adds tiff support" ON)
+option(USE_WEBP "adds webp support" ON)
+option(USE_LIBXML2 "adds libxml2 support" ON)
+option(USE_CAIRO "adds the cairo renderer" ON)
+option(USE_PROJ4 "adds proj4 support" ON)
+option(USE_GRID_RENDERER "adds grid renderer" ON)
+option(USE_SVG_RENDERER "adds grid renderer" ON)
+
+option(USE_PLUGIN_INPUT_CSV "adds plugin input csv" ON)
+option(USE_PLUGIN_INPUT_GDAL "adds plugin input gdal" ON)
+option(USE_PLUGIN_INPUT_GEOBUF "adds plugin input geobuf" ON)
+option(USE_PLUGIN_INPUT_GEOJSON "adds plugin input geojson" ON)
+option(USE_PLUGIN_INPUT_OGR "adds plugin input ogr" ON)
+option(USE_PLUGIN_INPUT_PGRASTER "adds plugin input pgraster" ON)
+option(USE_PLUGIN_INPUT_POSTGIS "adds plugin input postgis" ON)
+option(USE_PLUGIN_INPUT_RASTER "adds plugin input raster" ON)
+option(USE_PLUGIN_INPUT_SHAPE "adds plugin input shape" ON)
+option(USE_PLUGIN_INPUT_SQLITE "adds plugin input sqlite" ON)
+option(USE_PLUGIN_INPUT_TOPOJSON "adds plugin input topojson" ON)
+
+option(BUILD_DEMO_VIEWER "builds the demo viewer" ON)
+option(BUILD_DEMO_CPP "builds the demo c++ application" ON)
+
+option(BUILD_UTILITY_GEOMETRY_TO_WKB "builds the utility program geometry_to_wkb" ON)
+option(BUILD_UTILITY_MAPNIK_INDEX "builds the utility program mapnik_index" ON)
+option(BUILD_UTILITY_MAPNIK_RENDER "builds the utility program mapnik_render" ON)
+option(BUILD_UTILITY_OGRINDEX "builds the utility program ogrindex" OFF)
+option(BUILD_UTILITY_PGSQL2SQLITE "builds the utility program pgsql2sqlite" ON)
+option(BUILD_UTILITY_SHAPEINDEX "builds the utility program shapeindex" ON)
+option(BUILD_UTILITY_SVG2PNG "builds the utility program svg2png" ON)
+
+set(PLUGINS_INSTALL_DIR bin/plugins/input CACHE STRING "installs the plugins in the specified directory")
+get_filename_component(ABSOLUTE_PLUGINS_INSTALL_DIR ${PLUGINS_INSTALL_DIR} ABSOLUTE)
+message("Installing plugins to ${ABSOLUTE_PLUGINS_INSTALL_DIR}")
+set(FONTS_INSTALL_DIR bin/fonts CACHE STRING "installs the fonts in the specified directory")
+get_filename_component(ABSOLUTE_FONTS_INSTALL_DIR ${FONTS_INSTALL_DIR} ABSOLUTE)
+message("Installing fonts to ${ABSOLUTE_FONTS_INSTALL_DIR}")
+
```

Comment:
@mathisloge I had to add following (borrowed from TBB library) in order to build with `-stdlib=libc++` on Linux

```cmake
macro(CHECK_CXX_COMPILER_AND_LINKER_FLAGS _RESULT _CXX_FLAGS _LINKER_FLAGS)
  set(CMAKE_REQUIRED_FLAGS ${_CXX_FLAGS})
  set(CMAKE_REQUIRED_LIBRARIES ${_LINKER_FLAGS})
  set(CMAKE_REQUIRED_QUIET TRUE)
  check_cxx_source_runs("#include <iostream>\nint main(int argc, char **argv) { std::cout << \"test\"; return 0; }" ${_RESULT})
  set(CMAKE_REQUIRED_FLAGS "")
  set(CMAKE_REQUIRED_LIBRARIES "")
endmacro()

if (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
  if (CMAKE_CXX_FLAGS MATCHES "-stdlib=libc\\+\\+")
    message(STATUS "Mapnik: using libc++.")
  else()
    CHECK_CXX_COMPILER_AND_LINKER_FLAGS(HAS_LIBCPP "-stdlib=libc++" "-stdlib=libc++")
    if (HAS_LIBCPP)
      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -stdlib=libc++ -D_LIBCPP_VERSION")
      set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -stdlib=libc++")
      set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -stdlib=libc++")
      message(STATUS "Mapnik: using libc++.")
    else()
      message(STATUS "Mapnik: NOT using libc++.")
    endif()
  endif()
endif()

```
I don't know if this is a good way to do ?

Reply:
i usually prefer to using either a toolchain or directly the cmake arguments when configuring. 

But we could place a toolchain file in the cmake folder. which then could be used like that: 
cmake .. -DCMAKE_TOOLCHAIN_FILE=../cmake/clang.cmake 

This keeps the main cmake files free of ballast.

Reply:
Sure, `clang.cmake` sounds reasonable

Reply:
is added. 
I don't know if I'll get to it this weekend, I have to install ubuntu20 on another PC. But if you have some time, you can once `cmake ... -DCMAKE_TOOLCHAIN_FILE=../cmake/clang.cmake` during the initial configuration.

Reply:
:+1: , I'll give it a go, cheers!